plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -970,6 +970,14 @@ public void testCreatePartitionedTableWithNestedTypes()
         dropTable("test_partitioned_table_nested_type");
     }
 
+    @Test
+    public void testCreatePartitionedTableWithNestedField()
+    {
+        assertQueryFails(
+                "CREATE TABLE test_partitioned_table_nested_field(parent ROW(child VARCHAR)) WITH (partitioning = ARRAY['\"parent.child\"'])",
```

Comment:
Just to be on the safe side, we can also add an extra assertion with a 2 level nesting - grandparent.parent.child - I tried it and works as expected.